hardware/raspberrypi/bootmodes/gpio.md

```diff
@@ -0,0 +1,33 @@
+# GPIO boot mode
+
+The Raspberry Pi can be configured to allow the boot mode to be selected at power on using hardware attached to the GPIO connector: this is GPIO boot mode. This is done by setting bits in the OTP memory of the SoC. Once the bits are set, they permanently allocate 5 GPIOs to allow this selection to be made. Once the OTP bits are set they cannot be unset so you should think carefully about enabling this, since those 5 GPIO lines will then be unavailable for any other use.
+
+To enable GPIO boot mode, add the following line to the [config.txt](../../../configuration/config-txt/README.md) file:
+
+```
+program_gpio_bootmode=n
+```
+
+Where n is the bank of GPIOs which you wish to use. Then reboot the Pi once to program the OTP with this setting. Bank 1 is GPIOs 22-26, bank 2 is GPIOs 39-43. Unless you have a Compute Module, you must use bank 1: the GPIOs in bank 2 are only available on the Compute Module. Because of the way the OTP bits are arranged, if you first program GPIO boot mode for bank 1, you then have the option of selecting bank 2 later. The reverse is not true: once bank 2 has been selected for GPIO boot mode, you cannot select bank 1.
+
+Once GPIO boot mode is enabled, the Raspberry Pi will no longer boot. You must pull up at least one boot mode GPIO pin in order for the Pi to boot.
+
+Once enabled, the GPIOs are allocated to boot modes as follows:
+
+|Bank 1|Bank 2|boot type|
+|:----:|:---:|:--------:|
+|22    |39   |SD0       |
+|23    |40   |SD1       |
+|24    |41   |NAND (no Linux support at present)    |
+|25    |42   |SPI (no Linux support at present)    |
+|26    |43   |USB       |
+
```

Comment:
This is correct for Pi 3 but the following was added for Pi 3+

bank1   bank2   type
20         37         SD0
21         38         SD1
22         39         NAND
23         40         SPI
24         41         USB device
25         42         USB host - MSD
26         43         USB host - Ethernet